### ISSUE TITLE
About the GPU memory consumption.

### ISSUE THREAD

**Issue by AlbertHuyb** [liuzechun/ReActNet#26]
Thanks for your nice work. 
I'm trying to implement BNNs on other CV tasks utilizing your provided `BinaryActivation`, `LearnableBias` and `HardBinaryConv`, etc.

However, after replacing the normal operations with these modules, I noticed that the GPU memory consumption has greatly improved to perhaps 4.0x or even higher. Such a huge memory consumption makes it difficult to train with a large batch-size like 8.

Have you ever encountered similar problems? Do you have any ideas about it?

Thanks

**Comment by liuzechun** [comment]
Hi, thanks for your interest in our work! 

I think the memory increse main comes from the second-order activation binarization. Maybe you can try to change the L28 to L34 in https://github.com/liuzechun/ReActNet/blob/master/resnet/1_step1/birealnet.py to be the out3 = torch.clamp(x, -1, 1). This may yield slightly worse accuracy but save the memory.

Hope you find it useful.

**Comment by AlbertHuyb** [comment]
Thanks for your reply!
